blocks/rapid-result/rapid-result.js

```diff
@@ -0,0 +1,139 @@
+import { readBlockConfig, fetchPlaceholders, decorateIcons } from '../../scripts/aem.js';
+import { createElement, observe } from '../../scripts/blocks-utils.js';
+
+// TODO: This is dummy function that fetch sample data from EDS json.
+// It will be replaced when API call is available.
+async function fetchRapidResultMockData() {
+  let hostUrl = window.location.origin;
+  if (!hostUrl || hostUrl === 'null') {
+    // eslint-disable-next-line prefer-destructuring
+    hostUrl = window.location.ancestorOrigins[0];
+  }
+  const apiUrl = `${hostUrl}/draft/jiang/rapidresult.json`;
+  try {
+    const response = await fetch(apiUrl);
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    const results = data.data.map((result) => ({
+      title: result.title,
+      description: result.description,
+      link: result.link,
+      publishedon: result.publishedon,
+    }));
+    return results;
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.log('Failed to get API data: ', error);
+    return [];
+  }
+}
+
+function decorateTitle(titleContent) {
+  const title = createElement('div', 'title');
+  const h2 = createElement('h2', '');
+  h2.textContent = (titleContent ?? '').trim();
+  title.append(h2);
+  return title;
+}
+
+async function decorateCards(block, placeholders, previousNode) {
+  const results = await fetchRapidResultMockData();
+  const ul = createElement('ul', '');
+  // show 4 cards by default
```

Comment:
should we take this count `4` as a config input from the Word Document? wdyt

Reply:
number of cards to be shown can be calculated on the base of viewport/cardsize

Reply:
As confirmed with the customer, I'll make the total number of cards authorable. Regarding the block and card size, it doesn't show perfectly when there are more than 4 cards in a row. I'll keep the current site look & free, when the total card number is more than 4, it will be shown in the next row.